scipio/src/parking.rs

```diff
@@ -404,21 +414,29 @@ struct ReactorLock<'a> {
 
 impl ReactorLock<'_> {
     /// Processes new events, blocking until the first event or the timeout.
-    fn react(self, timeout: Option<Duration>) -> io::Result<()> {
-        // FIXME: there must be a way to avoid this allocation
-        // Indeed it just showed in a profiler. We can cap the number of
-        // cqes produced, but this is used for timers as well. Need to
-        // be more careful, but doable.
-        let mut wakers = Vec::new();
-
+    fn react(
+        self,
+        wakers: &mut Vec<Waker>,
+        timeout: Option<Duration>,
+        spin: bool,
+    ) -> io::Result<()> {
         // Process ready timers.
-        let next_timer = self.reactor.process_timers(&mut wakers);
+        let next_timer = self.reactor.process_timers(wakers);
 
         // Block on I/O events.
-        let res = match self.reactor.sys.wait(&mut wakers, timeout, next_timer) {
+        let ores = if spin {
+            self.reactor.sys.wait(wakers, next_timer, timeout)
```

Comment:
this definitely feels wrong... 
You are now passing the preempt timer as the `spin` parameter.